src/events/message-search.ts

```diff
@@ -0,0 +1,207 @@
+import { getLogger } from "../logger";
+import { addFunding, addNotice } from "../utils";
+
+// Only take certain plugins because we don't need to parse all markup like bolding
+// and the mention parsing is not as well-maintained as discord.js
+const ourRules = Object.fromEntries(
+	[
+		"escape", // type text, e.g. \`
+		"blockQuote", // type blockQuote, > OR >>>
+		"codeBlock", // type inlineCode, ```
+		"inlineCode", // type inlineCode, `
+		"spoiler", // type spoiler, ||
+		"text"
+	].map((label, order) => [label, { ...rules[label], order }])
+) as ParserRules;
+const parser = parserFor(ourRules);
+// Can improve in future to do the entire processing with this parser to just grab the search tokens we want
+
+// https://discord.com/developers/docs/reference#message-formatting-formats
+const mentionPatterns = (
+	["UserWithOptionalNickname", "Channel", "Role", "SlashCommand", "Emoji", "Timestamp"] as const
+).map(key => new RegExp(FormattingPatterns[key], "g"));
+
+export function cleanMessageMarkup(message: string): string {
+	// Remove the above markup elements
+	const nodes = parser(message, { inline: true });
+	message = nodes
+		.filter(node => node.type === "text")
+		.map(node => node.content)
+		.join("");
+	for (const regex of mentionPatterns) {
+		message = message.replaceAll(regex, "");
+	}
+	return message;
+}
+
+const DELIMITERS = {
+	ANGLE: { match: /<([^<\n]+?)>/g, prune: /<<.*?>>/g },
+	SQUARE: { match: /\[([^<\n]+?)\]/g, prune: /\[\[.*?\]\]/g },
+	CC: { match: /=([^<\n]+?)=/g, prune: /==.*?==/g },
+	TRANS: { match: /\)([^<\n]+?)\(/g, prune: null }, // ) (
+	HAVEN: { match: /%([^<\n]+?)\^/g, prune: null } // % ^
+} as const;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function getDelimiter(message: Message) {
+	switch (message.guildId) {
+		case "170669983079071745":
+			return DELIMITERS.CC;
+		case "597478047163219988":
+		case "845400480452050954":
+			return DELIMITERS.SQUARE;
+		case "871582695967301703":
+			return DELIMITERS.HAVEN;
+		case "780952232103116800":
+			return DELIMITERS.TRANS;
+		default:
+			return DELIMITERS.ANGLE;
+	}
+}
+
+export function parseSummons(cleanMessage: string, regex: RegExp): string[] {
+	return [...cleanMessage.matchAll(regex)]
+		.map(match => match[1].trim())
+		.filter(summon => {
+			// Ignore matches containing only whitespace or containing the following three tokens
+			if (summon.length === 0) {
+				return false;
+			}
+			const lower = summon.toLowerCase();
+			return !["://", "(", "anime"].some(token => lower.includes(token));
+		});
+}
+
+export function preprocess(
+	message: string,
+	delimiter: typeof DELIMITERS[keyof typeof DELIMITERS] = DELIMITERS.ANGLE
+): string[] {
+	message = cleanMessageMarkup(message);
+	if (delimiter.prune) {
+		message = message.replaceAll(delimiter.prune, "");
+	}
+	return parseSummons(message, delimiter.match);
+}
+
+/**
+ * Capture groups:
+ * - kid: % sign, if this matches a Konami ID, optional
```

Comment:
It doesn't hugely matter, because this isn't how we handle searches in slash commands anymore, but we used to use `#` to infer KIDs, right? I suppose this change from that is for lesser chance of conflict with existing card names?

Reply:
It's actually because <#1234> is a channel mention so it was never going to fly

Reply:
Oh, of course!

Reply:
If you have a better symbol idea, I'm all for it. We were stuck with the Discord mention conflict from the moment of original Bastion going with `<>` and becoming widely used 😛

Reply:
If I could go back and change one thing... No, I'm happy with this one! The fact it was a change just stood out to me.